dandiapi/api/views/users.py

```diff
@@ -15,6 +17,19 @@
 from dandiapi.api.permissions import IsApproved
 from dandiapi.api.views.serializers import UserDetailSerializer, UserSerializer
 
+logger = logging.getLogger(__name__)
+
+
+def _get_user_status(user: User):
+    # Workaround for https://github.com/dandi/dandi-archive/issues/1085
+    # TODO: remove/robustify some other way whenever underlying reason is
+    # identified
+    metadata = getattr(user, 'metadata', None)
+    if metadata:
+        return metadata.status
+    logger.error("User %s lacks .metadata. Returning user's status as INCOMPLETE", user)
+    return UserMetadata.Status.INCOMPLETE
```

Comment:
@djarecka make it here

```suggestion
    return UserMetadata.Status.APPROVED
```

if you want to allow that user to be able to click 'Create dataset' etc